# HG changeset patch
# User Chris Double <chris.double@double.co.nz>
# Date 1427767594 -46800
#      Tue Mar 31 15:06:34 2015 +1300
# Node ID 858ea94746eedc4b65734f8ceedfff7021940cb2
# Parent  1b8f36a7ee321ca62cf87424910d0e2ed6fed076
Bug 1148203 - Fix build errors with combinations of enable/disable of ffmp4/ffmpeg/eme - r=cpearce,ted

Fixes build error with  all three disabled. Prevents other combinations
that cause build errors by detecting them at configure time and displaying
an error message.

diff --git a/configure.in b/configure.in
--- configure.in
+++ configure.in
@@ -5316,6 +5316,10 @@ MOZ_ARG_DISABLE_BOOL(fmp4,
     MOZ_FMP4=,
     MOZ_FMP4=1)
 
+if test -n "$MOZ_FFMPEG" -a -z "$MOZ_FMP4"; then
+    AC_MSG_ERROR([Fragmented MP4 support must be enabled if using FFMPEG])
+fi
+
 if test -n "$MOZ_FMP4"; then
     AC_DEFINE(MOZ_FMP4)
     MOZ_EME=1
diff --git a/media/libstagefright/binding/MoofParser.cpp b/media/libstagefright/binding/MoofParser.cpp
--- media/libstagefright/binding/MoofParser.cpp
+++ media/libstagefright/binding/MoofParser.cpp
@@ -9,7 +9,7 @@
 
 #include "prlog.h"
 
-#ifdef PR_LOGGING
+#if defined(MOZ_FMP4) && defined(PR_LOGGING)
 extern PRLogModuleInfo* GetDemuxerLog();
 
 /* Polyfill __func__ on MSVC to pass to the log. */
